docs/updates/5-1-2017-flow.md

```diff
@@ -0,0 +1,156 @@
+### Flow Coverage Report (May 1st)
+
+`yarn flow-coverage`
+
+Stats:
+* Flow version `0.43.1`
+* **60 %**
+
+#### Components
+
+* **54%** coverage
+* 44 components are typed (almost all)
+* `yarn flow-react` - creates a flow report for components
+
+Typing our components helps us with rendering logic, which can take advantage
+of our domain objects such as `Frame`.
+
+```js
+function renderFrameTitle(frame: Frame) {
+  const displayName = formatDisplayName(frame);
+  return dom.div({ className: "title" }, displayName);
+}
+```
+
+We are not yet typing our props and state, which would be a helpful way
+to get additional coverage. We currently use `PropTypes`, which do not include
+flow types. We will hopefully switch to [props] soon to take advantage of static typing.
+
+```js
+shouldComponentUpdate(nextProps, nextState) {
+  const { frames, selectedFrame } = this.props;
+  const { showAllFrames } = this.state;
+  return (
+    frames !== nextProps.frames ||
+    selectedFrame !== nextProps.selectedFrame ||
+    showAllFrames !== nextState.showAllFrames
+  );
+}
+```
+
+There have been a couple places where flow types have been difficult to get right.
+
+1. **contravariants** `this.toggleFramesDisplay = this.toggleFramesDisplay.bind(this)`
+2. **defaultProps** and other class properties.
+
+Overall, typing our components has been a nice win though as it gives us additional confidence our UI will behave appropriately and catches several lifecycle edge-cases
+where properties could be null or racey.
```

Comment:
I think this means that some related data can be set in a non-deterministic order based on how async actions happen to fire. (Especially true for things like parsing that happen in a separate web worker thread.) Flow makes us declare which things have to be present together and prove that all possible state transitions meet those rules.